### ISSUE TITLE
Using multiple dataloaders at training time

### ISSUE THREAD

**Issue by dvirginz** [PyTorchLightning/pytorch-lightning#3282]
I Try to train on two dataloaders, one attached to a dataset where each `__get_item__` call fetches a predefined batch of varying length (thus the batch_size I transfer to the `dataloader` object is 1), and one where I sample randomly from a set of sequences, thus `__get_item__` call fetches one sample each time.

I'm looking for something like
```
            loader = DataLoader(
                batched_dataset,
                batch_size=1,
            )
            loader_tdm = DataLoader(
                random_samples_dataset,
                batch_size=8,
            )
            return [loader1,loader2]
```
I see this option is available for the validation and test dataloaders, while for training you suggested (previous issues) to use the `ConcatDataset`.
Is there a workaround?

**Comment by wolterlw** [comment]
Having a similar issue.
In my case I have two different data loaders - one loads video frames sequentially and the other loads individual frames using random sampling. The output format and tensors sizes are the same, but I need different samplers for each DataLoader.
I can create a complex sampler that knows which ordering to use for which subset of ConcatDataset, but that's error-prone.
Point being  - there seems to be a need to support multiple DataLoaders at training time

**Comment by wolterlw** [comment]
being resolved in #1959
